tests/helpers.py

```diff
@@ -0,0 +1,96 @@
+import pickle
+import unittest
+
+import glfw
+from mujoco_py.mjviewer import MjViewer
+import tensorflow as tf
+
+from tests.quirks import KNOWN_GYM_CLOSE_BROKEN
+from tests.quirks import KNOWN_GYM_NOT_CLOSE_VIEWER
+from tests.quirks import KNOWN_GYM_RENDER_NOT_IMPLEMENTED
+
+
+class TfTest(unittest.TestCase):
+    def setUp(self):
+        self.sess = tf.Session()
+        self.sess.__enter__()
+
+    def tearDown(self):
+        self.sess.__exit__(None, None, None)
+        self.sess.close()
+
+
+def step_env(env, n=10, render=True):
+    env.reset()
+    for _ in range(n):
+        _, _, done, _ = env.step(env.action_space.sample())
+        if render:
+            env.render()
+        if done:
+            break
+    env.close()
+
+
+def step_env_with_gym_quirks(test_case, env, spec, n=10, render=True):
+    env.reset()
+    for _ in range(n):
+        _, _, done, _ = env.step(env.action_space.sample())
+        if render:
+            if not spec.id in KNOWN_GYM_RENDER_NOT_IMPLEMENTED:
+                env.render()
+            else:
+                with test_case.assertRaises(NotImplementedError):
+                    env.render()
+        if done:
+            break
+    if not spec.id in KNOWN_GYM_CLOSE_BROKEN:
+        env.close()
+    else:
+        with test_case.assertRaisesRegex(
+                AttributeError, "'MjViewer' object has no attribute 'finish'"):
+            env.close()
+    if any([s in spec._entry_point for s in KNOWN_GYM_NOT_CLOSE_VIEWER]):
+        close_viewer_window(env)
+
+
+def pickle_env_with_gym_quirks(test_case, env, spec, render=True):
```

Comment:
I don't think this routine is necessary at all. It just pickles the environment followed by copy/paste of the helper above. Why not just write one after the other?

Reply:
I agree, however, I think adding a parameter in method step_env_with_gym_quirks to perform the pickle testing is a better option, since in these [lines](https://github.com/rlworkgroup/garage/blob/9f06a1e1754d73fac7ea8c077a77e60e147a8ef0/tests/helpers.py#L73-L74) I'm checking that the environment can be serialized once the viewer object is present (this is something important because it tests a bug we have already solved), but it has to be done before closing the environment, which is in the same block of code.